### PR TITLE
Revert to use of cos^2 in LiftDrag plugin

### PR DESCRIPTION
**Pull request by arjo129**
# 🦟 Bug fix

Fixes #<NUMBER>

## Summary
In response to https://github.com/gazebosim/gz-sim/pull/2189#issuecomment-1869798904

As suggested by @srmainwaring I also renamed cosSweepAngle ->
cos2SweepAngle to prevent further confusion.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by bperseghetti** [comment]
Looks like it needs a rebase and a commit squash to the 3 lines changed from cos->cos2.

**Comment by arjo129** [comment]
> Looks like it needs a rebase and a commit squash to the 3 lines changed from cos->cos2.

It's OK theres no need to do anything since we squash and rebase before we merge into main in anycase.

File: src/systems/lift_drag/LiftDrag.cc [hunk 1]
```diff
@@ -328,7 +328,7 @@ void LiftDragPrivate::Update(EntityComponentManager &_ecm)
       spanwiseI.Dot(velI), minRatio, maxRatio);
 
   // get cos from trig identity
```
Comment (srmainwaring) [inline src/systems/lift_drag/LiftDrag.cc:330]:
```suggestion
  // sweep adjustment depends on the velocity component which appears quadratically, so require cos^2.
```

**Review by srmainwaring — CHANGES_REQUESTED** [review]
LGTM. Suggested more detail in the comment to explain origin of formula.

**Review by srmainwaring — APPROVED** [review]

File: src/systems/lift_drag/LiftDrag.cc [hunk 2]
```diff
@@ -327,8 +327,9 @@ void LiftDragPrivate::Update(EntityComponentManager &_ecm)
   double sinSweepAngle = math::clamp(
       spanwiseI.Dot(velI), minRatio, maxRatio);
 
-  // get cos from trig identity
-  double cosSweepAngle = sqrt(1.0 - sinSweepAngle * sinSweepAngle);
+  // The sweep adjustment depends on the velocity component normal to the wing leading
```
Comment (azeey) [inline src/systems/lift_drag/LiftDrag.cc:330]: linter seems unhappy about these lines.

File: src/systems/lift_drag/LiftDrag.cc [hunk 3]
```diff
@@ -327,8 +327,9 @@ void LiftDragPrivate::Update(EntityComponentManager &_ecm)
...
+  // edge which appears quadratically in the dynamic pressure, so scale by cos^2 .
+  double cos2SweepAngle = 1.0 - sinSweepAngle * sinSweepAngle;
```
[the remark this answers is not in the document]
Comment (bperseghetti) [reply src/systems/lift_drag/LiftDrag.cc:332]: No, shouldn't be an issue for the PX4 models.

[the remark this answers is not in the document]
Comment (frede791) [reply src/systems/lift_drag/LiftDrag.cc:332]: No this is fine.

**Review by bperseghetti — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**